### ISSUE TITLE
Incorrect error note, possibly incorrect error altogether ?

### ISSUE THREAD

**Issue by carado** [rust-lang/rust#55486]
In both stable and the latest nightly, the following code:
```rust
trait Trait {}
impl<T> Trait for Box<T> {}
impl<T: 'static + Copy> Trait for T {}
```
Will fail with this message:
```
error[E0119]: conflicting implementations of trait `Trait` for type `std::boxed::Box<_>`:
 --> src/main.rs:3:1
  |
2 | impl<T> Trait for Box<T> {}
  | ------------------------ first implementation here
3 | impl<T: 'static + Copy> Trait for T {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `std::boxed::Box<_>`
  |
  = note: downstream crates may implement trait `std::marker::Copy` for type `std::boxed::Box<_>`
```
If I understand how traits work accross crate, the note on that error is false.

While I'm opening an issue about the erroneous note, I'm concerned that the error altogether might be mistaken. Is there a good reason for my example code to not be allowed ?

Thanks for your time.

**Comment by sfackler** [comment]
The error is present because you can't depend on the fact that an external type doesn't implement an external trait.

**Comment by carado** [comment]
Wouldn't it be reasonable to assume this isn't the case when both type and trait are not, as the note says, "downstream" ?
Then it would be impossible for a "downstream" crate to implement said trait for said type.

**Comment by sfackler** [comment]
Yep; this compiles, for example:

```rust
trait Foo {}

impl<T> Foo for Vec<T> {}

trait Bar {}

impl<T> Foo for T
where
    T: Bar
{}
```
